user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserQuotas.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.user.operator;
+
+import io.vertx.core.json.JsonObject;
+import org.I0Itec.zkclient.ZkClient;
+import org.I0Itec.zkclient.serialize.BytesPushThroughSerializer;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+/**
+ * Utility class for managing Quotas for Kafka User
+ */
+public class KafkaUserQuotas {
+    private static final Logger log = LogManager.getLogger(KafkaUserQuotas.class.getName());
+
+    private final static int CONNECTION_TIMEOUT = 30_000;
+
+    private ZkClient zkClient;
+
+    public KafkaUserQuotas(String zookeeperUrl, int zookeeperSessionTimeout) {
+        zkClient = new ZkClient(zookeeperUrl, zookeeperSessionTimeout, CONNECTION_TIMEOUT, new BytesPushThroughSerializer());
+    }
+
+    /**
+     * Create or update the quotas for the given user.
+     *
+     * @param username The name of the user which should be created or updated
+     * @param quotas The desired user quotas
+     */
+    public void createOrUpdate(String username, JsonObject quotas) {
+        byte[] data = zkClient.readData("/config/users/" + username, true);
+
+        if (data != null)   {
+            log.debug("Updating quotas for user {}", username);
+            zkClient.writeData("/config/users/" + username, updateUserJson(data, quotas));
```

Comment:
Should we try to diff this to make sure the we do not unnecessarily trigger an ZK update in broker?